### PR TITLE
[BugFix] Fix low cardinality columns not updated after data loading

### PR DESCRIPTION
**Pull request by abc982627271**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 16 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|16|00.00%|[67, 68, 91, 95, 96, 97, 99, 100, 134, 138, 139, 140, 141, 142, 144, 147]|

**Review by sduzh — DISMISSED** [review]

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java [hunk 1]
```diff
@@ -112,14 +131,20 @@ public void preWriteCommitLog(TransactionState txnState) {
         Preconditions.checkState(txnState.getTransactionStatus() == TransactionStatus.COMMITTED
                 || txnState.getTransactionStatus() == TransactionStatus.PREPARED);
         TableCommitInfo tableCommitInfo = new TableCommitInfo(table.getId());
+        boolean isFirstPartition = true;
         for (long partitionId : dirtyPartitionSet) {
             Partition partition = table.getPartition(partitionId);
-            long version = -1;
-            if (txnState.getTransactionStatus() == TransactionStatus.COMMITTED) {
-                version = partition.getNextVersion();
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java:119]: the code is used to fix transactional load in pr https://github.com/StarRocks/starrocks/pull/18158.

Comment (abc982627271) [reply fe/fe-core/src/main/java/com/starrocks/transaction/LakeTableTxnStateListener.java:119]: ok, I will rebase and commit it

**Review by sduzh — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 15 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/transaction/LakeTableTxnStateListener.java|0|15|00.00%|[67, 68, 91, 95, 96, 97, 99, 100, 134, 142, 143, 144, 145, 147, 150]|

**Comment by abc982627271** [comment]
run starrocks_admit_test

**Comment by abc982627271** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**